app/controllers/session_controller.rb

```diff
@@ -0,0 +1,21 @@
+class SessionController < ApplicationController
+  def new
+  end
+
+  def create
+    @user = User.find_by_username(params[:session][:username])
+    if @user && @user.authenticate(params[:session][:password])
+      session[:user_id] = @user.id
+      flash[:success] = "#{@user.username} has logged in"
+      redirect_to user_path @user.id
+    else
+      flash.now[:error] = "Invalid Credentials. Try again."
+      render :new
+    end
+  end
+
+  def destroy
+    session.destroy
+    redirect_to '/'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.